ArcGISRuntimeSDKQt_CppSamples/Maps/IdentifyLayers/IdentifyLayers.cpp

```diff
@@ -0,0 +1,157 @@
+// limitations under the License.
+// [Legal]
+
+#include "IdentifyLayers.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "Basemap.h"
+#include "ArcGISMapImageLayer.h"
+#include "ArcGISMapImageSublayer.h"
+#include "FeatureLayer.h"
+#include "ServiceFeatureTable.h"
+#include "Viewpoint.h"
+#include "Point.h"
+#include "IdentifyLayerResult.h"
+
+using namespace Esri::ArcGISRuntime;
+
+IdentifyLayers::IdentifyLayers(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void IdentifyLayers::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<IdentifyLayers>("Esri.Samples", 1, 0, "IdentifyLayersSample");
+}
+
+void IdentifyLayers::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a map using the topographic basemap
+  m_map = new Map(Basemap::topographic(this), this);
+
+  // add a map image layer
+  QUrl mapServiceUrl("http://sampleserver6.arcgisonline.com/arcgis/rest/services/SampleWorldCities/MapServer");
+  ArcGISMapImageLayer* imageLayer = new ArcGISMapImageLayer(mapServiceUrl, this);
+  m_map->operationalLayers()->append(imageLayer);
+
+  // hide 2 of the sublayers in the map image layer
+  connect(imageLayer, &ArcGISMapImageLayer::doneLoading, this, [this, imageLayer](Error e)
+  {
+    if (!e.isEmpty())
+    {
+      qDebug() << e.message() << e.additionalMessage();
+      return;
+    }
+
+    imageLayer->mapImageSublayers()->at(1)->setVisible(false);
+    imageLayer->mapImageSublayers()->at(2)->setVisible(false);
+  });
+
+  // add a feature layer
+  QUrl featureServiceUrl("https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0");
+  ServiceFeatureTable* featureTable = new ServiceFeatureTable(featureServiceUrl, this);
+  FeatureLayer* featureLayer = new FeatureLayer(featureTable, this);
+  m_map->operationalLayers()->append(featureLayer);
+
+  // set an initial viewpoint
+  double mapScale = 68015210;
+  double x = -10977012.785807;
+  double y = 4514257.550369;
+  Point pt(x, y, SpatialReference::webMercator());
+  Viewpoint initialViewpoint(pt, mapScale);
+  m_map->setInitialViewpoint(initialViewpoint);
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // connect signals
+  connectSignals();
+}
+
+void IdentifyLayers::connectSignals()
+{
+  // identify layers on mouse click
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent mouseEvent)
+  {
+    double tolerance = 12.0;
+    bool returnPopups = false;
+    int maxResults = 10;
+    m_mapView->identifyLayers(mouseEvent.x(), mouseEvent.y(), tolerance, returnPopups, maxResults);
+  });
+
+  // handle the identify results
+  connect(m_mapView, &MapQuickView::identifyLayersCompleted, this, [this](QUuid, QList<IdentifyLayerResult*> results)
```

Comment:
`const QList<IdentifyLayerResult*>& results` for the param to avoid a copy if signaling in the same thread.